From a474e6b97ccdf0900713bea7a2d1f71dbaf438c5 Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Tue, 15 May 2018 11:12:52 -0500
Subject: [PATCH] OS-6935 Possible race in lx_start_nfs_lockd Reviewed by:
 Jerry Jelinek <jerry.jelinek@joyent.com> Reviewed by: Mike Gerdts
 <mike.gerdts@joyent.com>

---
 usr/src/uts/common/brand/lx/os/lx_lockd.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/brand/lx/os/lx_lockd.c b/usr/src/uts/common/brand/lx/os/lx_lockd.c
index 234e815d3f..d6d965398a 100644
--- a/usr/src/uts/common/brand/lx/os/lx_lockd.c
+++ b/usr/src/uts/common/brand/lx/os/lx_lockd.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -77,6 +77,11 @@ lx_lockd_alive(pid_t lockd_pid)
 	}
 
 	mutex_enter(&p->p_lock);
+	if (p->p_stat == SZOMB || (p->p_flag & SEXITING) != 0) {
+		mutex_exit(&p->p_lock);
+		mutex_exit(&pidlock);
+		return (B_FALSE);
+	}
 	vp = p->p_exec;
 	VN_HOLD(vp);
 	mutex_exit(&p->p_lock);
-- 
2.21.0

